### PR TITLE
add missing types for PluginServiceStatic's methods

### PR DESCRIPTION
**Pull request by goonerholic**
Module: [Chart.js]()

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/chartjs/Chart.js/blob/2.9/src/core/core.plugins.js>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

**Review by lekoaf — CHANGES_REQUESTED** [review]
I took a look at the link you provided.

Did you not forget the `descriptors` function?

However, I'm not finding any of these functions in the official documentation. Could you point me in the right direction?

**Comment by goonerholic** [comment]
> I took a look at the link you provided.
> 
> Did you not forget the `descriptors` function?
> 
> However, I'm not finding any of these functions in the official documentation. Could you point me in the right direction?

Sorry, I forgot the method `descriptors`.

The documentation is somewhat unclear but these functions have been in the code since v.2.1 and hasn't been documented until `3.0.0-alpha`(https://www.chartjs.org/docs/3.0.0-alpha/typedoc/classes/_core_core_plugins_.pluginservice.html)

In the latest beta(3.0.0-beta7), PluginService totally changed and type definitions must be updated.
(https://www.chartjs.org/docs/next/typedoc/classes/_core_core_plugins_.pluginservice.html)

But current latest version is 2.9.4 and these missing types must be provided.

**Comment by lekoaf** [comment]
These type definitions will probably be obsolete when 3.0.0 is released. I think they are rewriting it with typescript? Or at least adding their own type defs.

**Comment by danmana** [comment]
While these methods are not documented on the docs site, I've found at least `Chart.plugins.getAll()` to be useful before:   
you can use it to [unregister a plugin that globally self registers](https://github.com/chartjs/Chart.js/issues/7963#issuecomment-717884919) by default, and re-register it per chart basis only where used

I don't see anything wrong with adding them (judging by the `@since` comments it looks like they might have been intended as a public API at least at some point)

Regarding chart.js 3.0, you are right the typings for it will be included in the [official package](https://github.com/chartjs/Chart.js/tree/master/types), so there will not be a need for them here.

File: types/chart.js/index.d.ts [hunk 1]
```diff
@@ -97,9 +97,20 @@ declare class Chart {
         [key: string]: Chart;
     };
 }
+type Plugin = Chart.PluginServiceGlobalRegistration & Chart.PluginServiceRegistrationOptions;
+interface PluginDescriptor {
+    plugins: Plugin;
```
Comment (danmana) [inline types/chart.js/index.d.ts:102]: I think this should be `plugin` not `plugins` based on imp https://github.com/chartjs/Chart.js/blob/2.9/src/core/core.plugins.js#L152
```suggestion
    plugin: Plugin;
```

**Review by danmana — CHANGES_REQUESTED** [review]

**Review by danmana — APPROVED** [review]

**Comment by goonerholic** [comment]
Ready to merge

**Comment by danmana** [comment]
Ready to merge

**Pull request merged**